autogen/agentchat/conversable_agent.py

```diff
@@ -785,13 +788,12 @@ def clear_history(self, recipient: Optional[Agent] = None, nr_messages_to_preser
         else:
             self._oai_messages[recipient].clear()
             if nr_messages_to_preserve:
-                print(
-                    colored(
+                iostream = self._iostream
+                with iostream.set_style(fg="yellow"):
+                    iostream.print(
```

Comment:
This is a pretty big deviation to how prints normally look in Python.

Why not create a class that matches the file interface (for outputting to other locations like a websocket) and pass that in to the file parameter of `print`? That way things still feel familiar

Reply:
That's an option I am considering, but in that case, it is more difficult to implement colouring and styles for non-console outputs. I am still not sure what is the best option so I am more than open to suggestions. This is a draft and just the current state of design space exploration.

Reply:
Totally, that's why I wanted to jump in :) So this seems like a decent option for getting back to a structured string from the formatted ansi escape codes https://github.com/getcuia/stransi

This would allow the websocket version to be a bit more structured about using it

Reply:
Cool! Let me try this out. Thanx :)

Reply:
@jackgerrits how about this? I'll deal with ANSI escape codes once I implement websockets.

Reply:
@jackgerrits I implemented websockets support without dealing with ANSI codes at all. Now I believe it should be handled on the client side and there are multiple options for handling it for each platform. For web based UI, it is really easy to find such components.

I believe the code is functionally correct as code coverage on tests is 100% and I really checked everything. Testing was quite tricky because I had to use sync version of the client and that meant running the server for testing in a separate thread. Async version would be much simpler and faster, but there is no easy way to mix sync and async code so had to stick with sync.

I have to write a notebook explaining how to set it up in practice.

Reply:
I feel like it's becoming more clear that the unstructured approach to output today is going to make building applications around this hard. Perhaps this is not to be dealt with in this PR but a raw stream of text is kind of hard to build a web app around

Reply:
We can deal with that at a later point though - there has been a desire for streaming responses so at least this unblocks that